edgehog-device-runtime-containers/src/docker/network.rs

```diff
@@ -0,0 +1,483 @@
+    Create(#[source] BollardError),
+    /// couldn't inspect the network
+    Inspect(#[source] BollardError),
+    /// couldn't remove the network
+    Remove(#[source] BollardError),
+    /// couldn't convert [`DockerNetwork`] into [`Network`]
+    Conversion(#[from] ConversionError),
+}
+
+/// Error converting [`DockerNetwork`] into [`Network`]
+#[non_exhaustive]
+#[derive(Debug, thiserror::Error, displaydoc::Display)]
+pub enum ConversionError {
+    /// missing network name or id
+    MissingName,
+}
+
+/// Container network struct.
+///
+/// Networks are user-defined networks that containers can be attached to.
+#[derive(Debug, Clone, Eq)]
+pub struct Network<S> {
+    /// Id of the container network.
+    pub id: Option<String>,
+    /// The network's name.
+    pub name: S,
+    /// The network's name.
+    /// Name of the network driver plugin to use.
+    ///
+    /// Defaults to "bridge"
+    pub driver: S,
+    /// Check for networks with duplicate names.
+    pub check_duplicate: bool,
+    /// Restrict external access to the network.
+    pub internal: bool,
+    /// Enable IPv6 on the network.
+    pub enable_ipv6: bool,
+}
+
+impl<S> Network<S> {
+    pub fn new(
+        id: Option<String>,
+        name: S,
+        driver: S,
+        check_duplicate: bool,
+        internal: bool,
+        enable_ipv6: bool,
+    ) -> Self {
+        Self {
+            id,
+            name,
+            driver,
+            check_duplicate,
+            internal,
+            enable_ipv6,
+        }
+    }
+
+    /// Get the network id or name if it's missing.
+    #[instrument]
+    pub fn network(&self) -> &str
+    where
+        S: AsRef<str> + Debug,
+    {
+        match &self.id {
+            Some(id) => {
+                trace!("returning id");
+
+                id.as_str()
+            }
+            None => {
+                trace!("id missing, returning name");
+
+                self.name.as_ref()
+            }
+        }
+    }
+
+    /// Set the id from docker.
+    fn update_id(&mut self, id: Option<String>)
+    where
+        S: Display,
+    {
+        if let Some(id) = id {
+            debug!("using id {id} for network {}", self.name);
+
+            let old_id = self.id.replace(id);
+
+            trace!(?old_id);
+        }
+    }
+
+    /// Check if the network exists if the id is set, otherwise it will try creating it.
+    ///
+    /// Returns true if the network was created.
+    pub async fn inspect_or_create(&mut self, client: &Client) -> Result<bool, NetworkError>
+    where
+        S: AsRef<str> + Display + Debug,
+    {
+        if self.id.is_some() {
+            match self.inspect(client).await? {
+                Some(net) => {
+                    trace!("found network {net:?}");
+
+                    return Ok(false);
+                }
+                None => {
+                    debug!("network not found, creating it");
+                }
+            }
+        }
```

Comment:
You can use let-else and remove the match statement.
```suggestion
            if let  Some(net) = self.inspect(client).await? {
                    trace!("found network {net:?}");
                    return Ok(false);
                }else {
                    debug!("network not found, creating it");
              }
        }
```

Reply:
this doesn't seem changed

Reply:
Mmm it did though, if you look at the log the comment is after the push. Maybe the page was outdated?